src/super_gradients/training/transforms/transforms.py

```diff
@@ -777,66 +800,30 @@ def __init__(
+        self.convert_format = DetectionFormatAdapter(input_format=self.input_format, output_format=self.output_format, image_shape=image_shape)
+        self.min_bbox_edge_size = self.min_bbox_edge_size / max(image_shape) if self.output_format.bboxes_format.format.normalized else self.min_bbox_edge_size
 
-        _, h, w = image.shape
-
-        def _format_target(targets_in):
-            if label_first_in_input:
-                labels, boxes = targets_in[:, 0], targets_in[:, 1:]
-            else:
-                boxes, labels = targets_in[:, :4], targets_in[:, 4]
-
-            if convert2cxcy:
-                boxes = xyxy2cxcywh(boxes)
-            elif convert2xyxy:
-                boxes = cxcywh2xyxy(boxes)
-
-            if normalize:
-                boxes[:, 0] = boxes[:, 0] / w
-                boxes[:, 1] = boxes[:, 1] / h
-                boxes[:, 2] = boxes[:, 2] / w
-                boxes[:, 3] = boxes[:, 3] / h
-
-            elif denormalize:
-                boxes[:, 0] = boxes[:, 0] * w
-                boxes[:, 1] = boxes[:, 1] * h
-                boxes[:, 2] = boxes[:, 2] * w
-                boxes[:, 3] = boxes[:, 3] * h
-
-            min_bbox_edge_size = self.min_bbox_edge_size / max(w, h) if normalized_output else self.min_bbox_edge_size
+    def __call__(self, sample):
+        def _format_target(targets: np.ndarray):
+            # INPUT FORMAT -> OUTPUT FORMAT
+            targets = self.convert_format(targets)
 
-            cxcywh_boxes = boxes if not output_xyxy_format else xyxy2cxcywh(boxes.copy())
+            # FILTER SMALL
+            def _is_too_small(bboxes: np.ndarray) -> np.ndarray:
+                return np.minimum(bboxes[:, 2], bboxes[:, 3]) > self.min_bbox_edge_size
 
-            mask_b = np.minimum(cxcywh_boxes[:, 2], cxcywh_boxes[:, 3]) > min_bbox_edge_size
-            boxes_t = boxes[mask_b]
-            labels_t = labels[mask_b]
+            targets = filter_on_bboxes(fn=_is_too_small, tensor=targets, tensor_format=self.output_format)
 
-            labels_t = np.expand_dims(labels_t, 1)
-            targets_t = np.hstack((labels_t, boxes_t)) if label_first_in_output else np.hstack((boxes_t, labels_t))
-            padded_targets = np.zeros((self.max_targets, 5))
-            padded_targets[range(len(targets_t))[: self.max_targets]] = targets_t[: self.max_targets]
+            # PAD
+            padded_targets = np.zeros((self.max_targets, targets.shape[-1]))
+            padded_targets[range(len(targets))[: self.max_targets]] = targets[: self.max_targets]
             padded_targets = np.ascontiguousarray(padded_targets, dtype=np.float32)
 
             return padded_targets
 
-        sample["target"] = _format_target(targets)
-        if crowd_targets is not None:
-            sample["crowd_target"] = _format_target(crowd_targets)
+        sample["target"] = _format_target(sample["target"])
+        if "crowd_target" in sample.keys():
```

Comment:
Shouldn't crowd targets be handled differently now that we can extend detection targets?
Lets at least put a TODO to refactor crowd targets.

Reply:
We could, but I think this would be an even bigger refacto, because it implies that the format of target/crowd_target is different either for the input, and/or for the output. In both cases, it means that the previous and/or following transforms should support target/crowrd_target with a different format, which is not the case.
But yeah I will write a ticket to think about this problematic